File: web/src/types/batch.ts
import { MedicineSummary } from "./medicine";

export interface BatchCreateRequest {
  medicine_id: number;
  batch_no: string;
  mfg_date: string; // YYYY-MM-DD
  exp_date: string; // YYYY-MM-DD
  quantity: number;
  mrp: number;
  status?: string;
}

export interface BatchResponse {
  id: number;
  medicine_id: number;
  batch_no: string;
  mfg_date: string;
  exp_date: string;
  quantity: number;
  mrp: number;
  status: string;
  created_at?: string | null;
}

export interface BatchDetailResponse extends BatchResponse {
  medicine?: MedicineSummary | null;
}

export interface CodeResponse {
  id: number;
  batch_id: number;
  serial_number: string;
  code_hash: string;
  qr_data_url?: string | null;
  qr_svg?: string | null;
  datamatrix_code?: string | null;
  status: string;
  scan_count: number;
  first_scanned_at?: string | null;
  created_at?: string | null;
}

export interface CodeGenerateBatchResponse {
  total_generated: number;
  batch_id: number;
  batch_no: string;
  medicine_name: string;
  codes: CodeResponse[];
}

File: backend/app/schemas/batch.py
from datetime import date, datetime
from typing import Optional
from pydantic import BaseModel, ConfigDict
from backend.app.schemas.medicine import MedicineSummary


class BatchBase(BaseModel):
    batch_no: str
    mfg_date: date
    exp_date: date
    quantity: int = 10000
    mrp: float = 50.0
    status: str = "active"


class BatchCreate(BatchBase):
    medicine_id: int


class BatchUpdate(BaseModel):
    batch_no: Optional[str] = None
    mfg_date: Optional[date] = None
    exp_date: Optional[date] = None
    quantity: Optional[int] = None
    mrp: Optional[float] = None
    status: Optional[str] = None


class BatchResponse(BatchBase):
    model_config = ConfigDict(from_attributes=True)

    id: int
    medicine_id: int
    created_at: Optional[datetime] = None


class BatchDetailResponse(BatchResponse):
    model_config = ConfigDict(from_attributes=True)

    medicine: Optional[MedicineSummary] = None
